sound/soc/sof/intel/hda-loader.c

```diff
@@ -503,12 +510,15 @@ int hda_dsp_post_fw_run_icl(struct snd_sof_dev *sdev)
 	 * the host whereas on TGL it is handled by the firmware.
 	 */
 	if (!hda->clk_config_lpro) {
-		ret = snd_sof_dsp_core_power_up(sdev, BIT(3));
+		ret = hda_dsp_enable_core(sdev, BIT(3));
 		if (ret < 0) {
 			dev_err(sdev->dev, "error: dsp core power up failed on core 3\n");
 			return ret;
 		}
 
+		sdev->enabled_cores_mask |= BIT(3);
+		sdev->dsp_core_ref_count[3]++;
+
 		snd_sof_dsp_stall(sdev, BIT(3));
```

Comment:
looks like we need a macro for this magic 3...

Reply:
It explains it, yes. And if it were just one (or two) use(s) of it, as it was until now, that would be ok-ish, but now I see 5 occurrences of "3" not counting the comment, and this is generic HDA code... So, I think it would be better to either move it to TGL-specific code - at least make a separate function for it, but even with a separate function, if it's kept in generic code, maybe it would be better to add a platform-specific bitmask and loop over bits in it... But it isn't critical at all, it just begins to look a bit fragile, error prone and too specific to me.

Reply:
I am going to leave this as 3 for now. But I will submit a separate PR to move this part of the code to icl.c and maybe replace the 3 with a macro there